libyul/optimiser/SSATransform.cpp

```diff
@@ -245,16 +246,18 @@ void IntroduceControlFlowSSA::operator()(Block& _block)
 				for (auto const& var: varDecl.variables)
 					if (m_variablesToReplace.count(var.name))
 					{
-						variablesDeclaredHere.insert(var.name);
-						m_variablesInScope.insert(var.name);
+						if (!util::contains(variablesDeclaredHere, var.name))
+							variablesDeclaredHere.push_back(var.name);
```

Comment:
Why not use the `emplaceBackUnique` here? As well in the other places that this pattern is used.
```suggestion
						util::emplaceBackUnique(variablesDeclaredHere, var.name);
```